### PR TITLE
Go Update

### PR DESCRIPTION
**Pull request by heidmotron**
Updated the golang metric registrar examples
* Updated the UI
* Changed the `custom` counter to a gauge
* Switched to go modules
* Added a manifest to make it easier to deploy

Screen Shot 2019-07-08 at 5 28 52 PM

### CODE REVIEW

File: golang/src/metric_registrar_examples/main.go [hunk 1]
```diff
@@ -1,69 +1,82 @@
...
-    time.Sleep(2 * time.Second)
-    w.Write([]byte("{}"))
+	time.Sleep(2 * time.Second)
+	w.Write([]byte("{}"))
 }
 
-func customMetric(w http.ResponseWriter, _ *http.Request) {
-    customCounter.Inc()
-    w.Write([]byte("{}"))
+func customMetric(w http.ResponseWriter, r *http.Request) {
```
Comment (josephgee) [inline golang/src/metric_registrar_examples/main.go]: The inverted condition here kind of obfuscates the fact that decrementing is a default case. Also, decrementing as a default case feels a little weird to me. 

Normally we'd expect a counter's value to monotonically increase, is decrementing even something we want to encourage?

Comment (josephgee) [reply golang/src/metric_registrar_examples/main.go]: On the Java app, you were decrementing the gauge but only incrementing the counter.

File: golang/src/metric_registrar_examples/metric_registrar_examples_test.go [hunk 2]
```diff
@@ -1,103 +1,115 @@
...
-    Expect(bodyBytes).To(BeEquivalentTo("{}"))
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(bodyBytes).To(BeEquivalentTo("{}"))
 }
 
 func startApp() *gexec.Session {
-    commandPath, err := gexec.Build(os.Getenv("GOPATH")+"/src/metric_registrar_examples", "-race")
-    Expect(err).ToNot(HaveOccurred())
+	commandPath, err := gexec.Build("./main.go", "-race")
```
Comment (josephgee) [inline golang/src/metric_registrar_examples/metric_registrar_examples_test.go]: Is there a reason to change this to a relative path?

File: golang/src/metric_registrar_examples/main.go (hunk 1 above)
Comment (heidmotron) [reply golang/src/metric_registrar_examples/main.go]: Ahh I should change the name, I changed to the type to a gauge but still called it a counter.

File: golang/src/metric_registrar_examples/metric_registrar_examples_test.go (hunk 2 above)
Comment (heidmotron) [reply golang/src/metric_registrar_examples/metric_registrar_examples_test.go]: Unless you set the GOPATH, there is no GOPATH, so when you push the app it actually can't load the UI.

Comment (josephgee) [reply golang/src/metric_registrar_examples/metric_registrar_examples_test.go]: Odd.K.

**Pull request merged**